src/AutoMapper/XpressionMapper/XpressionMapperVisitor.cs

```diff
@@ -231,9 +231,16 @@ protected void FindDestinationFullName(Type typeSource, Type typeDestination, st
             if (typeMap == null)
             {
                 throw QueryMapperHelper.MissingMapException(sourceType: typeDestination, destinationType: typeSource);
+            }
+
+            PathMap pathMap = typeMap.PathMaps.SingleOrDefault(item => string.Join(period, item.MemberPath.Members.Select(m => m.Name)) == sourceFullName);

```

Comment:
Maybe make this a TypeMap method?

Reply:
Less value here I think but no downside - done.

Reply:
Yes, probably no one else will use it, but it's no biggie.